spring-integration-file/src/main/java/org/springframework/integration/file/remote/handler/FileTransferringMessageHandler.java

```diff
@@ -139,7 +167,22 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.remoteFileTemplate.send(message, this.mode);
+		String path = this.remoteFileTemplate.send(message, this.mode);
+		if (this.chmod != null && isChmodCapable()) {
+			doChmod(this.remoteFileTemplate, path, this.chmod);
```

Comment:
I briefly thought about that; but figured it would most likely be a constant.

Let's wait to see if someone asks for it - I am not sure how an expression will help the octal conversion.